components/BlurryPoster/BlurryPoster.tsx

```diff
@@ -95,6 +109,53 @@ export const BlurryPoster = ({
   } = formatDate(publishedDate);
 
   let i = 1;
+  /**
+   * Foreground image/video
+   */
+  const hasVideo = !!videoWebm || !!videoMp4;
+  const hasMedia = !!imageSrc || hasVideo;
+
+  /**
+   * Background image/video
+   */
+  const hasBgVideo = !!bgVideoWebm || !!bgVideoMp4;
+  const bgVideoRef = useRef<HTMLVideoElement>(null);
+  const isBgVideoInView = useInView(bgVideoRef, { once: false, amount: 0.1 });
+  const [isBgPlaying, setIsBgPlaying] = useState<boolean>(false);
+  const [isBgUserPaused, setIsBgUserPaused] = useState<boolean>(false);
+
+  // Toggle foreground video play/pause
+  const toggleBgVideo = () => {
+    if (!bgVideoRef.current) return;
+
+    setIsBgPlaying((prev) => {
+      if (prev) {
+        bgVideoRef.current?.pause();
```

Comment:
do you need the `?` here as you've already checked if the `.current` value is truthy?

Reply:
Good catch! I'll update 😄